src/typechecker.erl

```diff
@@ -3812,6 +3833,8 @@ refinable(RefinableTy, Env, Trace)
                 not_found -> false
             end
     end;
+refinable(?type(range, _), _Env, _Trace) ->
+    true;
```

Comment:
Makes sense. No test for this? I would expect to see a known problem solved or a new test added.

Reply:
As above, it's tested by `test/should_fail/exhaustive_refinable_map_variants.erl`. Specifically, it's exercised by a `string()` being ultimately expanded to a `{type,0,list,[{type,0,range,[{integer,0,0},{integer,0,1114111}]}]}`.